### PR TITLE
Avoid TaskQueue allocations in Scheduler #30093

### PR DESCRIPTION
**Pull request by chessman**
References #30093

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

**Review by octonato — DISMISSED** [review]
~~LGTM~~

I dismissed my own review because I spotted something. See comments below.

File: akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala
Comment (octonato) [inline akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala:320]: In previous code we were creating a temporary `TaskQueue` to collect the nodes we wanted to put back. 

But here, we are adding all tasks back to the queue now. That same queue will be used again on next tick and will be carrying some baggage from the previous tick.

Comment (chessman) [reply akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala:320]: `spareTaskQueue` is an empty queue. We save all nodes we want to put back to this queue and put the queue back to the wheel. After we process all nodes from `tasks`, this queue is empty now and we reuse it as a new `spareTaskQueue` for the next wheel bucket. And so on, on each step we reuse an empty queue from the previous step.

Comment (octonato) [reply akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala:320]: Yes, you are perfectly right. Thanks.

**Review by octonato — APPROVED** [review]
LGTM 
(after second review and to the point explanation from @chessman)

File: akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala
Comment (johanandren) [inline akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala:320]:
```suggestion
        // we know tasks is empty now, so we can re-use it next tick
        spareTaskQueue = tasks
```

**Review by johanandren — APPROVED** [review]
LGTM, but good with a comment making it easier to get what's going on

**Review by patriknw — COMMENTED** [review]
late LGTM, thanks

**Pull request merged**
